### PR TITLE
swagger(iothub): Swagger comments for configuration APIs

### PR DESCRIPTION
**Pull request by vinagesh**

### CODE REVIEW

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 1]
```diff
@@ -11,8 +11,8 @@
   "paths": {
     "/configurations/{id}": {
       "get": {
-        "summary": "Retrieve a configuration for Iot Hub devices and modules by it identifier.",
-        "description": " For IoT Hub VNET related features(https://docs.microsoft.com/en-us/azure/iot-hub/virtual-network-support) please use API version '2020-03-13'.These features are currently in general availability in the East US, West US 2, and Southcentral US regions only. We are actively working to expand the availability of these features to all regions by end of month May. For rest of the APIs please continue using API version '2019-10-01'",
+        "summary": "Retrieve a configuration for Iot Hub device or module.",
+        "description": "Retrieve a configuration for Iot Hub device or modules.",
```
Comment (timtay-microsoft) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:15]: The summary and description here are redundant. When we generate code, only the description is used anywhere. I've been just deleting these summary tags when they don't have anything to say beyond what the description already says

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 2]
```diff
@@ -11,8 +11,8 @@
   "paths": {
     "/configurations/{id}": {
       "get": {
-        "summary": "Retrieve a configuration for Iot Hub devices and modules by it identifier.",
-        "description": " For IoT Hub VNET related features(https://docs.microsoft.com/en-us/azure/iot-hub/virtual-network-support) please use API version '2020-03-13'.These features are currently in general availability in the East US, West US 2, and Southcentral US regions only. We are actively working to expand the availability of these features to all regions by end of month May. For rest of the APIs please continue using API version '2019-10-01'",
+        "summary": "Retrieve a configuration for Iot Hub device or module.",
```
Comment (abhipsaMisra) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:14]: Retrieve a configuration for **an** Iot Hub device or module.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 1 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:15]: Yeah there are lots of those and I saw that only description is used during generation but did not delete to keep consistency. But if you are deleting them all over the place, i'll go ahead and do it too.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 3]
```diff
@@ -31,66 +32,69 @@
...
-              "$ref": "#/definitions/Configuration"
-            }
-          },
-          "201": {
-            "description": "Returns the created Configuration object",
-            "schema": {
-              "$ref": "#/definitions/Configuration"
+        "put": {
+            "summary": "Create or update the configuration for Iot Hub device or module.",
+            "description": "Create or update the configuration for Iot Hub device or module. ETag must not be specified for the create operation. ETag must be specified for the update operation. Configuration Id and Content cannot be updated.",
```
Comment (abhipsaMisra) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:44]: for this: Configuration Id and Content cannot be updated.
we will make this as readonly, no need to specify this in comments.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 4]
```diff
@@ -31,66 +32,69 @@
...
+                    "description": "The configuration to be created or updated.",
+                    "in": "body",
+                    "required": true,
+                    "schema": {
+                        "$ref": "#/definitions/Configuration"
+                    }
+                },
+                {
+                    "name": "If-Match",
+                    "description": "Header to make the request conditional. The operation is performed only if the ETag matches the value in the server. To force an unconditional update, set If-Match to the wildcard character (*).",
```
Comment (timtay-microsoft) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:71]: When describing the Etag, there is an RFC that is worth referencing. I made some mentions towards it in the registry APIs that I recommend you also use

Comment (abhipsaMisra) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:71]: small point, you might have already considered this - make sure this is the same as the description of etag for other update APIs as well

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 3 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:44]: Yes that is correct and will be in my next PR. I will remove it when I make that change. Keeping this now so that I can remind myself to make that change.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 5]
```diff
@@ -31,66 +32,69 @@
...
-            "schema": {
-              "$ref": "#/definitions/Configuration"
-            }
-          },
-          "201": {
-            "description": "Returns the created Configuration object",
-            "schema": {
-              "$ref": "#/definitions/Configuration"
+        "put": {
+            "summary": "Create or update the configuration for Iot Hub device or module.",
```
Comment (timtay-microsoft) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:43]: "... for your Iot Hub's devices and modules"?

Comment (timtay-microsoft) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:43]: As someone who isn't well versed in these APIs, being a bit more specific as to who all these configurations apply to seems helpful

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 4 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:71]: Good point. We might have to sync on this then as we are contributing per API. I'll make a task to standardize this.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 3 above)
Comment (abhipsaMisra) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:44]: I see, there is a potential for missing it, since those changes will be in a different file; make sure it is tracked properly.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 5 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:43]: Here is what I understood - You can update a Device or a module using this PUT. Is that incorrect and does the sentence not convey that?

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 3 above)
Comment (abhipsaMisra) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:44]: We don't want the service team to get the above comment as-is.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 6]
```diff
@@ -132,6 +138,7 @@
         "parameters": [
           {
             "name": "top",
+            "description": "Number of configurations to retrieve",
```
Comment (timtay-microsoft) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:141]: Do you think the service may override this value? We could ask them about this later if you leave a TODO here. ADT had something like this API where the service may override if you specify a number too large

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 7]
```diff
@@ -122,8 +128,8 @@
     },
     "/configurations": {
       "get": {
-        "summary": "Get multiple configurations for devices or modules of an IoT Hub. Returns the specified number of configurations for Iot Hub. Pagination is not supported.",
-        "description": " For IoT Hub VNET related features(https://docs.microsoft.com/en-us/azure/iot-hub/virtual-network-support) please use API version '2020-03-13'.These features are currently in general availability in the East US, West US 2, and Southcentral US regions only. We are actively working to expand the availability of these features to all regions by end of month May. For rest of the APIs please continue using API version '2019-10-01'",
+        "summary": "Get multiple configurations for Iot Hub devices or modules.",
+        "description": "Get multiple configurations for Iot Hub devices or modules.",
```
Comment (prmathur-microsoft) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:132]: Does ADM work on edge? or it specific only to IotHub?

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 8]
```diff
@@ -156,8 +163,8 @@
     },
     "/configurations/testQueries": {
       "post": {
-        "summary": "Validates the target condition query and custom metric queries for a configuration.",
-        "description": "Validates the target condition query and custom metric queries for a configuration. For IoT Hub VNET related features(https://docs.microsoft.com/en-us/azure/iot-hub/virtual-network-support) please use API version '2020-03-13'.These features are currently in general availability in the East US, West US 2, and Southcentral US regions only. We are actively working to expand the availability of these features to all regions by end of month May. For rest of the APIs please continue using API version '2019-10-01'",
+        "summary": "Validates the target condition and custom metric queries for a configuration.",
```
Comment (abhipsaMisra) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:166]: As Tim pointed out in the previous PR, summary tags are not being used anywhere in the autogen code; might be worth deleting them entirely.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 9]
```diff
@@ -143,7 +150,7 @@
         ],
         "responses": {
           "200": {
-            "description": "Returns the list of Configuration objects",
+            "description": "Returns the list of Configuration objects. Pagination is not supported.",
```
Comment (prmathur-microsoft) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:153]: Why not ask for feature request?

**Review by prmathur-microsoft — APPROVED** [review]

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 10]
```diff
@@ -494,8 +502,8 @@
         "parameters": [
           {
             "name": "id",
+            "description": "The device Id.",
```
Comment (abhipsaMisra) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:505]: This is worded as below, in the other APIs:
The unique identifier of the device to xxx.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 4 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:71]: Lets sync offline on this. I think I agree with Abhipsa that we should have the same description for this in all APIs.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 6 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:141]: I see. We can ask the service team about it.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 7 above)
Comment (abhipsaMisra) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:132]: We can have ADM for both devices and modules

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 8 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:166]: Yes I agree. I will delete for now. Do you guys know if there is any guidance to keep them at all even if they are redundant? My only thought is that the service team will generate this so we may just get them back based on the logic there.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 10 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:505]: Is that a common way of describing this?

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 9 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:153]: We are currently making SDK for existing IotHub features. The preview branch will have pnp stuff and anything new. We can surely ask them at that point. This change is mainly to send the service team content to update existing swagger comments. This will help up build without errors with amount of changes on our end. As for using the AsynPageable, we will not be able to use it if the service does not support. We can address the implementation part during API design. Note: This manual editing is not the approach we are taking to fix issues. This is just to send suggestions.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 2 above)
Comment (bikamani) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:14]: Shouldn't it be better to reword this as "Retrieve device or module configuration"?

Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:14]: Don't have strong opinions either way. I can reword to your suggestion.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 10 above)
Comment (timtay-microsoft) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:500]: It is, yes

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 11]
```diff
@@ -1799,53 +1802,53 @@
       "type": "object",
       "properties": {
         "id": {
-          "description": "Gets Identifier for the configuration",
+          "description": "The configuration Id.",
           "type": "string"
         },
         "schemaVersion": {
-          "description": "Gets Schema version for the configuration",
+          "description": "Schema version of configuration",
```
Comment (timtay-microsoft) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:1809]: "of the"

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 2 above)
Comment (timtay-microsoft) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: since it is a get on a configuration, I like Sindhu's current wording. The configuration sits above any individual device or module, so starting the sentence with "Retrieve a device or module..." would be confusing
